contrib/platform/src/com/sun/jna/platform/win32/WinGDI.java

```diff
@@ -28,18 +28,97 @@
 import com.sun.jna.Structure;
 import com.sun.jna.Structure.FieldOrder;
 import com.sun.jna.platform.win32.WinNT.HANDLE;
-import com.sun.jna.platform.win32.WinDef.HBITMAP;
-import com.sun.jna.platform.win32.WinDef.RECT;
+import com.sun.jna.Union;
+
+import static com.sun.jna.platform.win32.WinDef.*;
 
 /**
  * Ported from WinGDI.h.
  * Microsoft Windows SDK 6.0A.
  * @author dblock[at]dblock.org
  * @author Andreas "PAX" L&uuml;ck, onkelpax-git[at]yahoo.de
+ * @author SSHTOOLS Limited, support@sshtools.com
  */
 public interface WinGDI {
     int RDH_RECTANGLES = 1;
 
+    @FieldOrder({ "dmDeviceName", "dmSpecVersion", "dmDriverVersion", "dmSize", "dmDriverExtra", "dmFields", "dmUnion1", "dmColor",
+            "dmDuplex", "dmYResolution", "dmTTOption", "dmCollate", "dmFormName", "dmLogPixels", "dmBitsPerPel", "dmPelsWidth",
+            "dmPelsHeight", "dummyunionname2", "dmDisplayFrequency", "dmICMMethod", "dmICMIntent", "dmMediaType", "dmDitherType",
+            "dmReserved1", "dmReserved2", "dmPanningWidth", "dmPanningHeight" })
+
+    public static class DEVMODE extends Structure {
```

Comment:
I'm not sure if I should use `DEVMODE` or `DEVMODEA` here.  Some projects seem to have them [split out](https://github.com/redbooth/eclipse-platform-swt/blob/3e57ee0ed42c71cf5b75eff6d71364caee02a643/bundles/org.eclipse.swt/Eclipse%20SWT%20PI/win32/org/eclipse/swt/internal/win32/DEVMODEA.java#L13).  Advice appreciated.

Reply:
WIndows API often has a structure with different sizes based on ASCII (A) or UTF-16 Wide String (W).  So there's a `DEVMODEA` and `DEVMODEW` structure.  JNA's Windows API Type Mapper automatically picks the correct mapping for you if you use `DEVMODE`, and that will be the W structure for all modern versions of Windows.

But that means you need to give special treatment to arrays of characters.  The `dmDeviceName` field below is `byte[]` in the ASCII version and either `char[]` or `byte[]` with double the allocationin the Wide String version.   This is probably the part giving you problems.  Basically you can do something like this:
```java
public byte[] dmDeviceName = new byte[Winspool.CCHDEVICENAME * CHAR_WIDTH];`
```
And define CHAR_WIDTH as 1 for ASCII and 2 otherwise.   See this example:  https://github.com/java-native-access/jna/blob/fb76832972e92a215ecdb12ae4f3d6d0c2bae38d/contrib/platform/src/com/sun/jna/platform/win32/Wtsapi32.java#L231-L249

Reply:
I'll commit these proposed improvements, but the API still returns gibberish which changes each time I call it.  This may be a bug with JNA on Windows ARM64 that just-so-happens to surface in this test-case?  Perhaps it works better on an Intel machine. 😅 That's just a guess though.  Reference #1264.

Anyway, evidence of the gIbberish and raw hex bytes provided in two runs.  I'm using screenshots since the output is non-printable:

First run:
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/6345473/160659267-1cc355dd-2059-4e26-b658-55daf964906a.png">

Second run:
<img width="1083" alt="image" src="https://user-images.githubusercontent.com/6345473/160659579-3fc1e363-07c0-46ce-bd9e-c529a2ef0ece.png">

Reply:
@dbwiddis I'm marking this as resolved and moving my screenshots to the "gibberish" comment. :)